pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java

```diff
@@ -199,4 +207,47 @@ public static double jsonPathDouble(Object object, String jsonPath, double defau
       return defaultValue;
     }
   }
+
+  static class ArrayAwareJacksonJsonProvider extends JacksonJsonProvider {
+    @Override
+    public boolean isArray(Object obj) {
+      return (obj instanceof List) || (obj instanceof Object[]);
+    }
+
+    @Override
+    public Object getArrayIndex(Object obj, int idx) {
+      if (obj instanceof Object[]) {
+        return ((Object[]) obj)[idx];
+      }
+      return super.getArrayIndex(obj, idx);
+    }
+
+    @Override
+    public void setArrayIndex(Object obj, int index, Object newValue) {
```

Comment:
This is probably not required?

Reply:
looks like it's mainly for updating JSON object, like adding or setting a value at certain json path. Don't think it's needed for here, removing for now.